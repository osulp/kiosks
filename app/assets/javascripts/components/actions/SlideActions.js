export const SET_SLIDES = 'SET_SLIDES';
export const REFRESH_SLIDES = 'REFRESH_SLIDES';

export function setSlides(slides) {
  slides.push({id: 100, content: "absdg"});
  console.log(slides);
  return {
    type: SET_SLIDES,
    slides: slides
  };
}

export function refreshSlides(){
  // TODO: Ajax call to query the current set of slides?
}
