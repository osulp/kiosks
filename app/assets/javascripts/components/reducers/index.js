import { combineReducers } from 'redux';
import slidesReducer from './slidesReducer';

const rootReducer = combineReducers({
  slides: slidesReducer
});

export default rootReducer;