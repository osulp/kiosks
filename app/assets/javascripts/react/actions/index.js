export * from './kioskActions';
export * from './touchActions';
