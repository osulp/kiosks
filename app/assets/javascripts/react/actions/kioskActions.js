export const SET_KIOSK = 'SET_KIOSK';
export const setKiosk = (type, url) => {
  return {
    type: SET_KIOSK,
    kiosk: { type, url }
  };
};

export const SET_SLIDES = 'SET_SLIDES';
export const setSlides = (slides) => {
  return {
    type: SET_SLIDES,
    kiosk: {
      slides
    }
  };
};

export const SCROLL_TO_SLIDE = "SCROLL_TO_SLIDE";
export const scrollToSlide = (index) => {
  return {
    type: SCROLL_TO_SLIDE,
    index
  }
};

export const ADD_ERROR = 'ADD_ERROR';
export const addError = (error) => {
  return {
    type: ADD_ERROR,
    error
  };
};
