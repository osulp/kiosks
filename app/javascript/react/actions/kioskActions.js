export const SET_KIOSK = "SET_KIOSK"
export const setKiosk = (
  type,
  url,
  map_default_floor_number,
  maps_base_url,
  name,
  kid
) => {
  return {
    type: SET_KIOSK,
    kiosk: { type, url, map_default_floor_number, maps_base_url, name, kid }
  }
}

export const SET_SLIDES = "SET_SLIDES"
export const setSlides = slides => {
  return {
    type: SET_SLIDES,
    kiosk: {
      slides
    }
  }
}

export const SET_RESTART_KIOSK = "SET_RESTART_KIOSK"
export const setRestartKiosk = restart_kiosk => {
  return {
    type: SET_RESTART_KIOSK,
    kiosk: {
      restart_kiosk
    }
  }
}

export const SCROLL_TO_SLIDE = "SCROLL_TO_SLIDE"
export const scrollToSlide = index => {
  return {
    type: SCROLL_TO_SLIDE,
    index
  }
}

export const ADD_ERROR = "ADD_ERROR"
export const addError = error => {
  return {
    type: ADD_ERROR,
    error
  }
}

export const SET_TITLE = "SET_TITLE"
export const setTitle = title => {
  return {
    type: SET_TITLE,
    title
  }
}

export const SET_GOOGLE_ANALYTICS = "SET_GOOGLE_ANALYTICS"
export const setGoogleAnalytics = google_analytics => {
  return {
    type: SET_GOOGLE_ANALYTICS,
    google_analytics
  }
}
