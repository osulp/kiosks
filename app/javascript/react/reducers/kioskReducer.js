import {
  SCROLL_TO_SLIDE,
  SET_SLIDES,
  SET_RESTART_KIOSK,
  SET_KIOSK,
  ADD_ERROR,
  SET_TITLE,
  SET_GOOGLE_ANALYTICS
} from "../actions/kioskActions"

export const initial_state = {
  type: "touch",
  url: "kiosk/touch",
  api: {
    hours: "/api/v1/hours",
    available_rooms: "/api/v1/rooms/available/{date}"
  },
  title: "",
  slides: [],
  restart_kiosk: "",
  starting_slide_index: 0,
  errors: [],
  google_analytics: undefined
}

const kioskReducer = (state = initial_state, action) => {
  switch (action.type) {
    case ADD_ERROR:
      return Object.assign({}, state, {
        errors: [...state.errors, action.error]
      })
    case SET_KIOSK:
      return Object.assign({}, state, {
        type: action.kiosk.type,
        url: action.kiosk.url,
        map_default_floor_number: action.kiosk.map_default_floor_number,
        maps_base_url: action.kiosk.maps_base_url,
        name: action.kiosk.name,
        kid: action.kiosk.kid
      })
    case SET_RESTART_KIOSK:
      return Object.assign({}, state, {
        restart_kiosk: action.kiosk.restart_kiosk
      })
    case SET_SLIDES:
      return Object.assign({}, state, { slides: action.kiosk.slides })
    case SCROLL_TO_SLIDE:
      return Object.assign({}, state, { starting_slide_index: action.index })
    case SET_TITLE:
      return Object.assign({}, state, { title: action.title })
    case SET_GOOGLE_ANALYTICS:
      return Object.assign({}, state, {
        google_analytics: action.google_analytics
      })
    default:
      return state
  }
}

export default kioskReducer
