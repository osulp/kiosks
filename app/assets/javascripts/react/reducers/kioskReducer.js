import {SCROLL_TO_SLIDE, SET_SLIDES, SET_KIOSK, ADD_ERROR, SET_TITLE, SET_GOOGLE_ANALYTICS} from '../actions/kioskActions';

export const initial_state = {
  type: "touch",
  url: "kiosk/touch",
  api: {
    hours: "/api/v1/hours",
    classroom_schedule: "/api/v1/classrooms/date/{date}",
    classrooms: "/api/v1/classrooms/rooms",
    available_rooms: "/api/v1/rooms/available/{date}"
  },
  title: "Donor Impact",
  slides: [],
  starting_slide_index: 0,
  errors: [],
  google_analytics: undefined
};

const kioskReducer = (state = initial_state, action) => {
  switch (action.type) {
    case ADD_ERROR:
      return Object.assign({}, state, { errors: [...state.errors, action.error] });
    case SET_KIOSK:
      return Object.assign({}, state, { type: action.kiosk.type, url: action.kiosk.url });
    case SET_SLIDES:
      return Object.assign({}, state, { slides: action.kiosk.slides });
    case SCROLL_TO_SLIDE:
      return Object.assign({}, state, { starting_slide_index: action.index });
    case SET_TITLE:
      return Object.assign({}, state, { title: action.title });
    case SET_GOOGLE_ANALYTICS:
      return Object.assign({}, state, { google_analytics: action.google_analytics });
    default:
      return state;
  }
};

export default kioskReducer;
