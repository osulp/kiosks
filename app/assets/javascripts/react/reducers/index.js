import { combineReducers } from 'redux';
import kioskReducer from './kioskReducer';
import modalReducer from './modalReducer';
import touchReducer from './touchReducer';
import circReducer from './circReducer';

export const REDUCERS_SHAPE = {
  kiosk: kioskReducer,
  modal: modalReducer,
  touch: touchReducer,
  circ: circReducer
};

export default combineReducers(REDUCERS_SHAPE);
