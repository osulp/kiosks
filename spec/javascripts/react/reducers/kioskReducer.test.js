import {default as reducerCreator} from '../../../../app/assets/javascripts/react/reducers/kioskReducer';
import * as actionCreator from '../../../../app/assets/javascripts/react/actions/kioskActions';
import * as factories from '../.factories';

const empty_state = undefined;

describe('Reducers::Kiosk', () => {
  it('matches the snapshot', () => {
    expect(reducerCreator).toMatchSnapshot();
  });
  describe('addError action', () => {
    it('matches the snapshot', () => {
      expect(reducerCreator(empty_state, actionCreator.addError(factories.error))).toMatchSnapshot();
    })
  });
  describe('setKiosk action', () => {
    it('matches the snapshot', () => {
      expect(reducerCreator(empty_state, actionCreator.setKiosk(factories.kiosk))).toMatchSnapshot();
    })
  });
  describe('setSlides action', () => {
    it('matches the snapshot', () => {
      expect(reducerCreator(empty_state, actionCreator.setSlides(factories.slides))).toMatchSnapshot();
    })
  });
  describe('setRestartKiosk action', () => {
    it('matches the snapshot', () => {
      expect(reducerCreator(empty_state, actionCreator.setRestartKiosk(factories.kiosk.restart_kiosk))).toMatchSnapshot();
    })
  });
  describe('scrollToSlide action', () => {
    it('matches the snapshot', () => {
      expect(reducerCreator(empty_state, actionCreator.scrollToSlide(1))).toMatchSnapshot();
    })
  });
  describe('setTitle action', () => {
    it('matches the snapshot', () => {
      expect(reducerCreator(empty_state, actionCreator.setTitle("MyTitle"))).toMatchSnapshot();
    })
  });
});
