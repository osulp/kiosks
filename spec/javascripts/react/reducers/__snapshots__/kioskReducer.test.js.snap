exports[`Reducers::Kiosk addError action matches the snapshot 1`] = `
Object {
  "api": Object {
    "available_rooms": "/api/v1/rooms/available/{date}",
    "hours": "/api/v1/hours",
  },
  "errors": Array [
    Object {
      "code": "500",
      "message": "Error",
    },
  ],
  "google_analytics": undefined,
  "restart_kiosk": "",
  "slides": Array [],
  "starting_slide_index": 0,
  "title": "",
  "type": "touch",
  "url": "kiosk/touch",
}
`;

exports[`Reducers::Kiosk matches the snapshot 1`] = `[Function]`;

exports[`Reducers::Kiosk scrollToSlide action matches the snapshot 1`] = `
Object {
  "api": Object {
    "available_rooms": "/api/v1/rooms/available/{date}",
    "hours": "/api/v1/hours",
  },
  "errors": Array [],
  "google_analytics": undefined,
  "restart_kiosk": "",
  "slides": Array [],
  "starting_slide_index": 1,
  "title": "",
  "type": "touch",
  "url": "kiosk/touch",
}
`;

exports[`Reducers::Kiosk setKiosk action matches the snapshot 1`] = `
Object {
  "api": Object {
    "available_rooms": "/api/v1/rooms/available/{date}",
    "hours": "/api/v1/hours",
  },
  "errors": Array [],
  "google_analytics": undefined,
  "kid": undefined,
  "map_default_floor_number": undefined,
  "maps_base_url": undefined,
  "name": undefined,
  "restart_kiosk": "",
  "slides": Array [],
  "starting_slide_index": 0,
  "title": "",
  "type": Object {
    "api": Object {
      "classroom_schedule": "/api/v1/classrooms/date/{date}",
      "classrooms": "/api/v1/classrooms/rooms",
      "hours": "/api/v1/hours",
    },
    "errors": Array [],
    "restart_kiosk": "",
    "slides": Array [],
    "starting_slide_index": 0,
    "title": "",
    "type": "touch",
    "url": "/url/to/action",
  },
  "url": undefined,
}
`;

exports[`Reducers::Kiosk setRestartKiosk action matches the snapshot 1`] = `
Object {
  "api": Object {
    "available_rooms": "/api/v1/rooms/available/{date}",
    "hours": "/api/v1/hours",
  },
  "errors": Array [],
  "google_analytics": undefined,
  "restart_kiosk": "",
  "slides": Array [],
  "starting_slide_index": 0,
  "title": "",
  "type": "touch",
  "url": "kiosk/touch",
}
`;

exports[`Reducers::Kiosk setSlides action matches the snapshot 1`] = `
Object {
  "api": Object {
    "available_rooms": "/api/v1/rooms/available/{date}",
    "hours": "/api/v1/hours",
  },
  "errors": Array [],
  "google_analytics": undefined,
  "restart_kiosk": "",
  "slides": Array [
    Object {
      "caption": "Slide.caption",
      "created_at": "2000-12-31T11:59:59Z",
      "expires_at": "2020-12-31T23:59:59Z",
      "id": 1,
      "kiosk": "Kiosk.name",
      "original": "/uploads/default_slide_1.png",
      "slide_type": "SlideType.name",
      "thumbnail": "/uploads/default_slide_1.png",
      "title": "Slide.title",
      "updated_at": "2000-12-31T23:59:59Z",
      "xlarge": "/uploads/default_slide_1.png",
    },
    Object {
      "caption": "Slide.caption:2",
      "created_at": "2000-11-11T11:00:00Z",
      "expires_at": "2010-11-11T13:00:00Z",
      "id": 2,
      "kiosk": "Kiosk.name:2",
      "original": "/uploads/default_slide_2.png",
      "slide_type": "SlideType.name:2",
      "thumbnail": "/uploads/default_slide_1.png",
      "title": "Slide.title:2",
      "updated_at": "2000-11-11T13:00:00Z",
      "xlarge": "/uploads/default_slide_1.png",
    },
  ],
  "starting_slide_index": 0,
  "title": "",
  "type": "touch",
  "url": "kiosk/touch",
}
`;

exports[`Reducers::Kiosk setTitle action matches the snapshot 1`] = `
Object {
  "api": Object {
    "available_rooms": "/api/v1/rooms/available/{date}",
    "hours": "/api/v1/hours",
  },
  "errors": Array [],
  "google_analytics": undefined,
  "restart_kiosk": "",
  "slides": Array [],
  "starting_slide_index": 0,
  "title": "MyTitle",
  "type": "touch",
  "url": "kiosk/touch",
}
`;
